app/src/main/java/com/infomaniak/drive/ui/fileList/FileAdapter.kt

```diff
@@ -241,50 +249,104 @@ open class FileAdapter(
-                checkIfEnableFile(file)
+            if (progress != Utils.INDETERMINATE_PROGRESS || !file.isPendingOffline(binding.context)) {
+                fileItemViews.apply {
+                    progressLayoutView.setupFileProgress(file, true)
+                    checkIfEnableFile(file, binding.context)
+                }
             }
         } else if (payloads.firstOrNull() is Boolean) {
-            val isChecked = payloads.first() as Boolean
-            val isGrid = viewHolderType == DisplayType.GRID
-            fileChecked.isChecked = isChecked
-            fileChecked.isVisible = isGrid || isChecked
-            filePreview.isInvisible = (isChecked && !isGrid) || getFile(position).isImporting()
+            fileItemViews.toggleFileCheckedState(isFileChecked = payloads.first() as Boolean, position)
         } else {
             super.onBindViewHolder(holder, position, payloads)
         }
     }
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) = with(holder.itemView.fileCardView) {
-        if (getItemViewType(position) != VIEW_TYPE_LOADING) {
-            val file = getFile(position)
-            val isGrid = viewHolderType == DisplayType.GRID
+    private fun FileItemViews.toggleFileCheckedState(isFileChecked: Boolean, position: Int) {
+        val isGrid = viewHolderType == DisplayType.GRID
 
-            if (isGrid) {
-                if (isHomeOffline) {
-                    (layoutParams as ConstraintLayout.LayoutParams).apply {
-                        dimensionRatio = null
-                        height = 150.toPx()
-                    }
+        fileChecked.apply {
+            isChecked = isFileChecked
+            isVisible = isGrid || isFileChecked
+        }
+        filePreview.isInvisible = (isFileChecked && !isGrid) || getFile(position).isImporting()
+    }
+
+    override fun onBindViewHolder(holder: FileViewHolder, position: Int) = with(holder) {
+        val viewType = getItemViewType(position)
+        if (viewType == VIEW_TYPE_LOADING) return@with
+
+        val file = getFile(position)
+        val isGrid = viewHolderType == DisplayType.GRID
+        val fileCardView = fileItemViews.cardView
+
+        if (isGrid) {
+            if (isHomeOffline) {
+                (fileCardView.layoutParams as ConstraintLayout.LayoutParams).apply {
+                    dimensionRatio = null
+                    height = 150.toPx()
                 }
-            } else {
-                setCorners(position, itemCount)
             }
+        } else {
+            fileCardView.setCorners(position, itemCount)
+        }
+
+        when (viewType) {
+            DisplayType.LIST.layout -> (binding as CardviewFileListBinding).itemViewFile.setFileItem(file, isGrid)
+            DisplayType.GRID.layout -> (binding as CardviewFileGridBinding).setFileItem(file, isGrid)
+            DisplayType.GRID_FOLDER.layout -> (binding as CardviewFolderGridBinding).setFileItem(file, isGrid)
+        }
```

Comment:
Would making `setFileItem` an extension of FileItemViews be possible? We'd need some more views inside the data class but we could avoid this `when` which feels like the goal of this "common views interface" through which we interact

Reply:
Doing this would need to expose the interface publicly, which I wanted to avoid, and thus instantiate it in other places. 
Furthermore I like that the differentiated setFileItem allows a "strong typed"  way for the function, as we sure to understand what code if for what binding. Otherwise I would need to pass 3 or 4 more nullable views in the data class which would mitigate its lisibility and need
So finally this when is a lot clearer than the other solution

Reply:
Yeah no this is indeed used in many places, I think you've got the best solution